clients/src/main/java/org/apache/kafka/common/serialization/ListSerializer.java

```diff
@@ -46,9 +51,14 @@ public ListSerializer() {}
     public ListSerializer(Serializer<Inner> serializer) {
         this.inner = serializer;
         this.isFixedLength = serializer != null && fixedLengthSerializers.contains(serializer.getClass());
+        this.serStrategy = this.isFixedLength ? SerializationStrategy.NULL_INDEX_LIST : SerializationStrategy.NEGATIVE_SIZE;
```

Comment:
By default, if we are dealing with a list of primitives, we are using `SerializationStrategy.NULL_INDEX_LIST` vs. a list of non-primitives (`UUID`, `String`, or some custom object) `SerializationStrategy.NEGATIVE_SIZE`.